packages/valory/skills/market_creation_manager_abci/behaviours.py

```diff
@@ -2031,6 +2078,263 @@ def _mark_market_as_done(
+        )
+        if response.status_code != HTTP_OK:
+            self.context.logger.error(
+                f"Could not retrieve response from {self.params.newsapi_endpoint}."
+                f"Received status code {response.status_code}.\n{response}"
+            )
+            return None
+
+        response_data = json.loads(response.body.decode())
+        self.context.logger.info(
+            f"Response received from {self.params.newsapi_endpoint}:\n {response_data}"
+        )
+        return response_data["articles"]
+
+    def _get_answer_tx(
+        self, question_id: str, answer: str
+    ) -> Generator[None, None, Optional[Dict[str, Any]]]:
+        """Get an answer a tx."""
+        response = yield from self.get_contract_api_response(
+            performative=ContractApiMessage.Performative.GET_STATE,  # type: ignore
+            contract_address=self.params.realitio_contract,
+            contract_id=str(RealtioContract.contract_id),
+            contract_callable="get_submit_answer_tx",
+            question_id=bytes.fromhex(question_id[2:]),
+            answer=bytes.fromhex(answer[2:]),
+            max_previous=MAX_PREVIOUS,
+        )
+        if response.performative != ContractApiMessage.Performative.STATE:
+            self.context.logger.error(
+                f"Couldn't get submitAnswer transaction. "
+                f"Expected response performative {ContractApiMessage.Performative.STATE.value}, "  # type: ignore
+                f"received {response.performative.value}."
+            )
+            return None
+
+        data = cast(bytes, response.state.body["data"])
+        return {
+            "to": self.params.realitio_contract,
+            "value": self.params.close_question_bond,
+            "data": data,
+        }
+
+    def get_payload(self) -> Generator[None, None, str]:
+        """Get the transaction payload"""
+        # get the questions to that need to be answered
+        questions = yield from self._get_unanswered_questions()
+        if questions is None:
+            self.context.logger.info("Couldn't get the questions")
+            return CloseMarketsRound.ERROR_PAYLOAD
+
+        if len(questions) == 0:
+            self.context.logger.info("No questions to close")
+            return CloseMarketsRound.NO_TX
+
+        self.context.logger.info(
+            f"Got {len(questions)} questions to close. " f"Questions: {questions}"
+        )
+
+        # get the answers for those questions
+        question_to_answer = {}
+        for question in questions:
+            answer = yield from self._get_answer(question["title"])
+            if answer is None:
+                self.context.logger.warning(
+                    f"Couldn't get answer for question {question}"
+                )
+                continue
+            question_to_answer[question["question"]["id"]] = answer
+
+            if len(question_to_answer) == self.params.num_questions_to_close:
```

Comment:
What is this doing ?

Reply:
Limiting the number of questions we answer at once.

Reply:
Ok, the name is a bit misleading, i thougt it was the total number of questions to answer overall.

Reply:
I'll rename to `questions_to_close_batch_size`, ok?

Reply:
done